### PR TITLE
Configuration to create single coverage report for all librarie.js tests

### PR DESCRIPTION
**Pull request by monikaprabhu**
- Move mocha test files to folder `__ tests __` so all tests can be run by jest using single command and fix the file path to `__tests__/mochatest/libraryUtilitiesTest.ts`
- Add config variable `collectCoverageFrom` to jest in package.json to filter the files tracked by jest cover. 
- Modify config variable `testRegex` to `/__tests__/*/.*` in package.json to include test files in nested folder.
- Move `libraryUtilitiesTestClasses.ts` to folder `__tests__/data` and jest excludes this from running.
- Now package.json inlcudes a variable `testPathIgnorePatterns` for jest to exclude test data folder `__tests__/data/`

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### CODE REVIEW

**Comment by monikaprabhu** [comment]
@fanwgwg please review and merge. 

fyi... @Benglin , @ellensi

File: package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "scripts": {
     "test": "mocha -r ts-node/register test/libraryUtilitiesTest.ts",
```
Comment (fanwgwg) [inline package.json]: This line should be changed to the new locations of the mocha tests

Comment (Benglin) [inline package.json]: @fanwgwg do you think we can merge `test`, `testjest` and `testcover` into one command `test`? Either case, we need `ReadMe.md` updated.

File: package.json [hunk 2]
```diff
@@ -50,7 +51,7 @@
     "transform": {
       ".(ts|tsx)": "<rootDir>/node_modules/ts-jest/preprocessor.js"
     },
-    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(ts|tsx|js)$",
+    "testRegex": "(/__tests__/.*|/*|\\.(test|spec))\\.(ts|tsx|js)$",
```
Comment (Benglin) [inline package.json]: Can we briefly describe this change? I'm trying to understand more

File: package.json (hunk 1 above)
Comment (Benglin) [inline package.json]: Also, are those underscores required in folders like `__tests__` and `__mocks__`?

Comment (monikaprabhu) [inline package.json]: @Benglin This is to include nested folders in "__tests__" folder.  I've fixed a bug in this line in last commit in this PR.

File: package.json (hunk 2 above)
Comment (monikaprabhu) [inline package.json]: @Benglin This is to include nested folders in "__tests__" folder while running tests.  I've fixed a bug in this line in last commit in this PR.

**Comment by monikaprabhu** [comment]
@fanwgwg , @Benglin please review and merge. 
@ellensi fyi..

File: package.json (hunk 1 above)
Comment (monikaprabhu) [inline package.json]: @fanwgwg I've fixed the path for mocha tests.

**Comment by fanwgwg** [comment]
@monikaprabhu @Benglin I think it is possible to combine `test`, `testjest` and `testcover` all to one command. The current behaviour is that running `testcover` will run all the jest test, since those mocha test are in the same folder as the jest test, those mocha test will be executed as well. Therefore, running `testcover` could run all the test and print a coverage report. 
I would suggest that we just leave one line for test in the scripts, which is `"test": "jest --coverage"`.

**Comment by monikaprabhu** [comment]
@monikaprabhu @Benglin regarding combining of commands : 
"test" is for non UI tests, if we don't want to run non UI tests separately we can remove this option. 
"testjest" is for all tests. Let's keep this option. 
"testcover" is for CI to create the coverage report. I don't think its necessary to create coverage every time developer run tests. Let me know if you agree.

**Comment by monikaprabhu** [comment]
@Benglin ,@fanwgwg - I've removed the three test commands. There is only one test command now, which will run all tests and create a coverage report.

**Comment by monikaprabhu** [comment]
@Benglin , @fanwgwg - I had to revert two commits as they were made on wrong version of pacakge.json. 
The last commit removes the test commands. Now only one test command "npm run test" remains which will run all tests and create a coverage report.

**Comment by monikaprabhu** [comment]
@Benglin - all comments above are resolved - this can be merged now.

File: __tests__/jesttest.tsx [hunk 3]
```diff
@@ -6,6 +6,10 @@ import * as LibraryEntryPoint from '../src/entry-point';
 import LibraryContainer from '../src/components/LibraryContainer';
 import LibraryItem from '../src/components/LibraryItem';
 
+
+// To run all tests with coverage report use command "npm run test"
+// To run jest tests without coverage report use command "npm run jest"
+
```
Comment (Benglin) [inline __tests__/jesttest.tsx]: These descriptions are no longer valid, and also they need to be moved under `ReadMe.md` if it isn't already there (we won't have test instructions inside a test file in general).

File: __tests__/mochatest/libraryUtilitiesTest.ts [hunk 4]
```diff
@@ -1,9 +1,12 @@
...
-import * as testClasses from './libraryUtilitiesTestClasses'
+import * as LibraryUtilities from '../../src/LibraryUtilities'
+import * as testClasses from '../data/libraryUtilitiesTestClasses'
 
 import { expect } from 'chai';
 import 'mocha';
 
+// To run all tests with coverage report use command "npm run test"
+// To run mocha tests use command "npm run test mocha test/libraryUtilitiesTest.ts"
+
```
Comment (Benglin) [inline __tests__/mochatest/libraryUtilitiesTest.ts]: Please also move these description to `ReadMe.md` file.

File: package.json [hunk 5]
```diff
@@ -3,8 +3,7 @@
   "description": "Project that contains all hosted contents of Dynamo Windows client",
   "main": "index.js",
   "scripts": {
-    "test": "mocha -r ts-node/register test/libraryUtilitiesTest.ts",
-    "testjest": "jest",
+    "test": "jest --coverage",
```
Comment (Benglin) [inline package.json]: Thanks, this is good! 👍

File: package.json [hunk 6]
```diff
@@ -35,6 +34,7 @@
     "cross-env": "^3.2.4",
     "css-loader": "^0.27.3",
     "enzyme": "^2.8.2",
+    "enzyme-to-json": "^1.5.1",
```
Comment (Benglin) [inline package.json]: Hi @monikaprabhu, does it work if we don't have this dependency?

File: package.json (hunk 1 above)
Comment (Benglin) [inline package.json]: @monikaprabhu do you mean we now have `tests` and `mocks` folders instead of `__tests__` and `__mocks__`?

**Comment by monikaprabhu** [comment]
@Benglin - removed comments and removed the entry for "enzyme-to-json": "^1.5.1". Please review and merge

**Comment by monikaprabhu** [comment]
@fanwgwg can you please review this ?

**Comment by fanwgwg** [comment]
@monikaprabhu Should we update the readme about the tests and also the coverage report?

**Comment by monikaprabhu** [comment]
@fanwgwg - I've updated Readme. Please review and merge

File: README.md [hunk 7]
```diff
@@ -23,15 +23,10 @@ Installing all dependencies
 ## Run Tests
 - Run UI test
```
Comment (fanwgwg) [inline README.md]: We don't need to specify that this is UI test anymore since we are combining all the tests in one command.

**Comment by monikaprabhu** [comment]
@fanwgwg , done.

**Comment by fanwgwg** [comment]
@monikaprabhu This looks good to me!

**Pull request merged**